oci/archive/oci_transport.go

```diff
@@ -0,0 +1,199 @@
+package archive
```

Comment:
(Not reviewing the naming and the like. at all, I assume that will mirror the outcome of #318.)